src/controller/CHIPDeviceController.cpp

```diff
@@ -1733,6 +1733,51 @@ void DeviceCommissioner::OnDone(app::ReadClient *)
             return CHIP_NO_ERROR;
         }
     });
+
+    // Try to parse as much as we can here before returning, even if this is an error.
+    return_err = (err == CHIP_NO_ERROR) ? return_err : err;
+
+    err = mAttributeCache->ForEachAttribute(
+        app::Clusters::OperationalCredentials::Id, [this, &info](const app::ConcreteAttributePath & path) {
+            if (path.mAttributeId != app::Clusters::OperationalCredentials::Attributes::Fabrics::Id)
+            {
+                // Continue on
+                return CHIP_NO_ERROR;
+            }
+
+            // this code is checking if the device is already on the commissioner's fabric.
+            // if a matching fabric is found, then remember the nodeId so that the commissioner
+            // can cancel commissioning (before it fails in AddNoc) and know it's nodeId.
+            switch (path.mAttributeId)
+            {
+            case app::Clusters::OperationalCredentials::Attributes::Fabrics::Id: {
+                app::Clusters::OperationalCredentials::Attributes::Fabrics::TypeInfo::DecodableType fabrics;
+                ReturnErrorOnFailure(
+                    this->mAttributeCache->Get<app::Clusters::OperationalCredentials::Attributes::Fabrics::TypeInfo>(path,
+                                                                                                                     fabrics));
+                auto iter = fabrics.begin();
+                while (iter.Next())
+                {
+                    auto & fabricDescriptor = iter.GetValue();
+                    ChipLogProgress(AppServer,
+                                    "DeviceCommissioner::OnDone - fabric.vendorId=0x%04X fabric.fabricId=0x" ChipLogFormatX64
+                                    " fabric.nodeId=0x" ChipLogFormatX64,
+                                    fabricDescriptor.vendorId, ChipLogValueX64(fabricDescriptor.fabricId),
+                                    ChipLogValueX64(fabricDescriptor.nodeId));
+                    if (GetFabricId() == fabricDescriptor.fabricId)
+                    {
+                        ChipLogProgress(AppServer, "DeviceCommissioner::OnDone - found a matching fabric");
+                        info.nodeId = fabricDescriptor.nodeId;
```

Comment:
This does not look right.  This is just checking the integer fabric id and ignoring the scoping root public key, no?

Reply:
Yes, agreed. Discussed it offline with @chrisdecenzo and made that point and gave pointers how to resolve

Reply:
good catch - added code to also check the fabric root public key